first_order_model/train.py

```diff
@@ -201,19 +202,126 @@ def train(config, generator, discriminator, kp_detector, checkpoint, log_dir, da
     generator_full = GeneratorFullModel(kp_detector, generator, discriminator, train_params)
     discriminator_full = DiscriminatorFullModel(kp_detector, generator, discriminator, train_params)
     
+    for x in dataloader:
+        if use_lr_video or use_RIFE:
+            lr_frame = F.interpolate(x['driving'], lr_size)
+            if train_params.get('encode_video_for_training', False):
+                target_bitrate = train_params.get('target_bitrate', None)
+                quantizer_level = train_params.get('quantizer_level', -1)
+                if target_bitrate == 'random':
+                    target_bitrate = np.random.randint(15, 75) * 1000
+                nr = x.get('time_base_nr', torch.ones(lr_frame.size(dim=0), dtype=int))
```

Comment:
This whole segment should just involve one call to a function under my revamped main. https://github.com/geminovc/model/blob/ae031b220e1eafca16f82fec0416154a937ac35a/first_order_model/utils.py#L52. Please refactor